### DISCUSSION TITLE
Expected date for Release containing Rack 3.1 Compatibility?

### DISCUSSION THREAD

**Discussion by pJeyakumar** [sinatra/sinatra#2040] [Q&A] [answered]
Hello,

I was wondering what the expected release date for the Sinatra Release containing [this patch](https://github.com/sinatra/sinatra/pull/2035) for Rack 3.1 compatibility will be?

Our team has Sinatra as a dependency for one of our applications and we do plan on upgrading to Rack 3.2 (once it is released), as such we'd like to resolve all warnings (specifically the `lib/sinatra/base.rb:1849: warning: Rack::Logger is deprecated and will be removed in Rack 3.2.`).

Thank you for all the work you do, and I look forward to hearing from you soon!

**Comment by dentarg** [comment, marked as answer]
I think we are looking at October if we're realistic

**Comment by dentarg** [comment]
[4.1.0](https://github.com/sinatra/sinatra/pull/2063) (and [4.1.1](https://github.com/sinatra/sinatra/pull/2068)) has been released now
